app/src/main/java/org/dieschnittstelle/mobile/android/skeleton/TaskDetailViewActivity.java

```diff
@@ -78,13 +86,30 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
         this.viewModel.isTaskOnDelete().observe(this, onDelete -> {
             if (onDelete) {
-                Intent returnIntent = new Intent();
-                returnIntent.putExtra(TASK_DETAIL_VIEW_KEY, task);
-                this.setResult(TaskDetailViewActivity.RESULT_OK, returnIntent);
-                this.finish();
+                AlertDialog.Builder builder = new AlertDialog.Builder(this);
+                builder.setView(android.R.layout.select_dialog_item);
+                builder.setMessage("Delete task?");
+                builder.setPositiveButton("Delete", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int id) {
+                        deleteIntent();
+                    }
+                });
+                builder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int id) {
+                        // User cancels the dialog.
+                    }
+                });
+                AlertDialog dialog = builder.create();
+                dialog.show();
```

Comment:
ich würde die Erstellung von AlertDialog als eigene Funktion definieren, wie beim DueDate und TimeLimit unten

Reply:
Ist in eigener Methode jetzt, musste aber deleteIntent extra erstellen, ansonsten kein Zugriff auf Instanzmethoden